services/gitdiff/git_diff_tree.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2025 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package gitdiff
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+)
+
+type DiffTree struct {
+	Files []*DiffTreeRecord
+}
+
+type DiffTreeRecord struct {
+	// Status is one of 'added', 'deleted', 'modified', 'renamed', 'copied', 'typechanged', 'unmerged', 'unknown'
+	Status string
+
+	// For renames and copies, the percentage of similarity between the source and target of the move/rename.
+	Score uint8
+
+	HeadPath   string
+	BasePath   string
+	HeadMode   git.EntryMode
+	BaseMode   git.EntryMode
+	HeadBlobID string
+	BaseBlobID string
+}
+
+// GetDiffTree returns the list of path of the files that have changed between the two commits.
+// If useMergeBase is true, the diff will be calculated using the merge base of the two commits.
+// This is the same behavior as using a three-dot diff in git diff.
+func GetDiffTree(ctx context.Context, gitRepo *git.Repository, useMergeBase bool, baseSha, headSha string) (*DiffTree, error) {
```

Comment:
In below code, `GetCommit` will convert any "ref" to a commit, so is it intentionally to support "baseSha=main" (branch/tag names) here?